sql/moz-fx-data-shared-prod/udf/pseudonymize_ad_id/udf.sql

```diff
@@ -0,0 +1,23 @@
+-- pseudonymize_ad_ids: Only for use in Ad ID pipeline
+CREATE OR REPLACE FUNCTION udf.pseudonymize_ad_id(hashed_ad_id STRING, key BYTES)
+RETURNS STRING AS (
+  IF( -- These two values represent the hashed all-0 Ad IDs, see bug 1823482
+    hashed_ad_id IN (
+      "q7RZx9leE4AHjn9DRQR8kLtjO0FNwmj/61ECGxFinno=",
+      "ksEEnM6/esBxl2myqrVLo31ePBoQZZnkCm81vefaF90="
+    ),
+    NULL,
+    TO_HEX(`moz-fx-data-shared-prod`.udf.hmac_sha256(key, CAST(ad_id AS BYTES)))
```

Comment:
```suggestion
    TO_HEX(`moz-fx-data-shared-prod`.udf.hmac_sha256(key, CAST(hashed_ad_id AS BYTES)))
```

Reply:
hmm these tests must have not run or they'd have failed

Reply:
CI did fail

Reply:
So it did!